### PR TITLE
[flutter_tools] fix --observatory-port not working

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

https://github.com/flutter/flutter/issues/69722

This is only a fix for master/dev/beta. Stable will need a separate cherry pick

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 1]
```diff
@@ -1486,7 +1486,7 @@ description: Dart SDK extensions for dart:ui
 homepage: http://flutter.io
 # sky_engine requires sdk_ext support in the analyzer which was added in 1.11.x
 environment:
-  sdk: '>=1.11.0 <3.0.0'
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/commands/update_packages.dart:1489]: I'm guessing this in an opportunistic change. Want to do this in a separate commit in case this gets rolled back, and to not confuse a stable cherry pick attempt?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/update_packages.dart:1489]: Oops, I did not mean to commit this - but I was originally using this branch for something else 😓

**Comment by jonahwilliams** [comment]
@jmagman I don't think we can/should CP because I could not repro on the previous stable. I wonder if the issue there is specific to iOS?

**Review by jmagman — APPROVED** [review]
LGTM

**Comment by jason-simmons** [comment]
I thought `--observatory-port` was supposed to be deprecated.

Is the issue that `--observatory-port` should now map to `--dds-port` instead of `--host-vmservice-port`?  The `--dds-port` is used to serve the observatory UI and is probably what users intend to control if they specify an `--observatory-port`.

@bkonyi

**Comment by jonahwilliams** [comment]
ultimately the plumbing is the same, so either --observatory-port or --host-vmservice-port will work.

Fixing for real needs some more thought, but we can't ship a stable with this broken

**Comment by jonahwilliams** [comment]
@bkonyi can you confirm that switching observatory port to dds port is a safe change? If so, we should probably remove the --dds-port option instead of disabling it

**Comment by bkonyi** [comment]
No, we need to be able to specify the VM service and DDS ports separately to handle remote workflows. `--observatory-port` should continue to function the way it always has. See my follow up on the issue.
